components/button-group/button-group.js

```diff
@@ -0,0 +1,162 @@
+import { LocalizeCoreElement } from '../../lang/localize-core-element.js';
+import { offscreenStyles } from '../offscreen/offscreen.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+/**
+ *
+ * A button group component that can be used to display a set of buttons
+ *
+ * @slot - Buttons, dropdown buttons, links or other items to be added to the container
+ */
+class ButtonGroup extends RtlMixin(LocalizeCoreElement(DynamicResizeMixin(LitElement))) {
+
+	static get properties() {
+		return {
+			/**
+			 * Setting this property to true will change the look of the dopdown button to a subtle button for subtle button groups
+			 */
+			subtle: {
+				type: Boolean,
+				attribute: 'subtle',
+			},
+
+		};
+	}
+
+	static get styles() {
+		return [offscreenStyles,
+			css`
+			:host {
+				display: block;
+
+			}
+			
+			:host([subtle]) {
+				--d2l-button-icon-fill-color: var(--d2l-color-celestine);
+				--d2l-button-icon-fill-color-hover: var(--d2l-color-celestine-minus-1);
+			}
+
+			:host .d2l-button-group-container {
+				display: flex;
+				flex: 0 1 auto;
+				flex-wrap: wrap;
+			}
+
+			:host .d2l-button-group-container ::slotted(d2l-button),
+			:host .d2l-button-group-container ::slotted(d2l-button-icon),
+			:host .d2l-button-group-container ::slotted(d2l-link),
+			:host .d2l-button-group-container ::slotted(span),
+			:host .d2l-button-group-container ::slotted(d2l-dropdown:not(.d2l-overflow-dropdown)),
+			:host .d2l-button-group-container ::slotted(d2l-dropdown-button),
+			:host .d2l-button-group-container ::slotted(d2l-dropdown-button-subtle),
+			:host .d2l-button-group-container ::slotted(.d2l-button-group-custom-item) {
+				margin-right: 0.6rem;
+			}
+
+			:host([dir="rtl"]) .d2l-button-group-container ::slotted(d2l-button),
+			:host([dir="rtl"]) .d2l-button-group-container ::slotted(d2l-button-icon),
+			:host([dir="rtl"]) .d2l-button-group-container ::slotted(d2l-link),
+			:host([dir="rtl"]) .d2l-button-group-container ::slotted(span),
+			:host([dir="rtl"]) .d2l-button-group-container ::slotted(d2l-dropdown:not(.d2l-overflow-dropdown)),
+			:host([dir="rtl"]) .d2l-button-group-container ::slotted(d2l-dropdown-button),
+			:host([dir="rtl"]) .d2l-button-group-container ::slotted(d2l-dropdown-button-subtle),
+			:host([dir="rtl"]) .d2l-button-group-container ::slotted(.d2l-button-group-custom-item) {
+				margin-left: 0.6rem;
+				margin-right: 0;
+			}
+
+			:host([subtle]) .d2l-button-group-container ::slotted(d2l-button-subtle) {
+				margin-right: 0.2rem;
+			}
+
+			:host([subtle][dir="rtl"]) .d2l-button-group-container ::slotted(d2l-button-subtle) {
+				margin-left: 0.2rem;
+				margin-right: 0;
+			}
+
+			:host([subtle]) .d2l-dropdown-subtle-opener-text,
+			.d2l-dropdown-opener-text {
+				margin-right: 0.3rem;
+				vertical-align: middle;
+			}
+			:host(:dir(rtl)) .d2l-dropdown-opener-text,
+			:host([subtle]:dir(rtl)) .d2l-dropdown-subtle-opener-text {
```

Comment:
You'll want `:host([dir="rtl"])` not the `:dir(rtl)` as the `RtlMixin` will sprout the `dir` attribute on the host for you.